girder/models/collection.py

```diff
@@ -114,22 +115,33 @@ def createCollection(self, name, creator, description='', public=True):
         :type public: bool
         :param creator: The user who is creating this collection.
         :type creator: dict
+        :param reuseExisting: If a collection with the given name already exists
+            return that collection rather than creating a new one.
+        :type reuseExisting: bool
         :returns: The collection document that was created.
         """
+        if reuseExisting:
+            existing = self.findOne({
+                'name': name
+            })
+            if existing:
+                return existing
+
         now = datetime.datetime.utcnow()
 
         collection = {
             'name': name,
             'description': description,
-            'creatorId': creator['_id'],
+            'creatorId': creator['_id'] if creator else None,
```

Comment:
This introduces the notion of collections with no creator, so we'll definitely want to add some testing since that's a new feature.